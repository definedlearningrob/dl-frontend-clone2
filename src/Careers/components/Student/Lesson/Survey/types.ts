export type SURVEY_QUESTION_TYPE = 'single_choice' | 'multiple_choice';

type CareerReviewSurveyQuestionOption = {
  option: string;
  step: number;
};

type CareerReviewSurveyQuestion = {
  answer: string[];
  id: string;
  options: CareerReviewSurveyQuestionOption[];
  question: string;
  type: SURVEY_QUESTION_TYPE;
};

export type CareerReviewSurvey = {
  __typename: string;
  questions: CareerReviewSurveyQuestion[];
  performed: boolean;
  version: number;
};
