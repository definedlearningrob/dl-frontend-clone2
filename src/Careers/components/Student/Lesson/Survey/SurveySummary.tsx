import { useTranslation } from 'react-i18next';
import { isEmpty } from 'lodash-es';
import cx from 'classnames';

import { TCareerReviewSurveyQuestion } from '@dc/resources/types';

type Props = {
  questions: TCareerReviewSurveyQuestion[];
};

export const SurveySummary = ({ questions }: Props) => {
  const { t } = useTranslation();

  return (
    <>
      <div className='mb-base'>
        <h4 className='mb-xs'>{t('careerReviewSurvey.previousAnswers')}</h4>
        <p className='text-xs xxxl:text-sm leading-lg'>{t('careerReviewSurvey.summaryInfo')}</p>
      </div>
      {questions.map(({ answer, question }, questionIndex) => {
        const questionLabel = `${questionIndex + 1}. ${question}`;

        const questionLabelClassName = cx('mb-0', { ['mb-xs']: !isEmpty(answer) });

        return (
          <div
            key={`question-${question}`}
            className='rounded-sm p-sm border border-neutral-300 bg-neutral-200 mb-base last-of-type:mb-0'>
            <h6 className={questionLabelClassName}>{questionLabel}</h6>
            <ul className='list-disc pl-base marker:text-neutral-700 text-xs'>
              {answer.map((answer) => (
                <li key={`answer-${answer}`} className='leading-lg'>
                  {answer}
                </li>
              ))}
            </ul>
          </div>
        );
      })}
    </>
  );
};
