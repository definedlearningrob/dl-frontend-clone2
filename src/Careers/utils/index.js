const isObject = (object) => object !== null && typeof object === 'object';

export const isEqual = (object1, object2) => {
  if (object1 === object2) return true;

  if (
    object1 === null ||
    object2 === null ||
    typeof object1 !== 'object' ||
    typeof object2 !== 'object'
  )
    return false;

  const keys1 = Object.keys(object1);
  const keys2 = Object.keys(object2);

  if (keys1.length !== keys2.length) {
    return false;
  }

  for (let key of keys1) {
    const val1 = object1[key];
    const val2 = object2[key];
    const areObjects = isObject(val1) && isObject(val2);
    if (
      // eslint-disable-next-line
      areObjects && !isEqual(val1, val2) ||
      // eslint-disable-next-line
      !areObjects && val1 !== val2
    ) {
      return false;
    }
  }

  return true;
};

export const getByStringKey = (object, key) => {
  const keys = key.split('.');

  return keys.reduce((acc = {}, key) => acc[key], object);
};
