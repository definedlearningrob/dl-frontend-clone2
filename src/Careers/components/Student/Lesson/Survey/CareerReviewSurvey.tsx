import { Formik } from 'formik';
import { castArray, isEmpty, map } from 'lodash-es';
import { useHistory, useParams } from 'react-router-dom';
import { useTranslation } from 'react-i18next';
import { ApolloError } from '@apollo/client';

import { ReactComponent as TableOfContentIcon } from '@dc/svg/table_of_content.svg';
import { SurveyForm } from '@dc/components/Student/Lesson/Survey/Form/SurveyForm';
import { OPPORTUNITY_QUERY } from '@dc/graphql/student/queries/opportunity';
import { VIRTUAL_INTERNSHIP_CONTENT_QUERY } from '@dc/graphql/student/queries/virtualInternshipContent';
import currentCoursesQuery from '@dc/graphql/student/queries/currentCourses';
import { useCreateCareerReviewSurveyAnswers } from '@dc/graphql/student/hooks/useCreateCareerReviewSurveyAnswers';
import { CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES } from '@dc/resources/enums';
import { SurveySummary } from '@dc/components/Student/Lesson/Survey/SurveySummary';
import { TCareerReviewSurvey } from '@dc/components/Student/Lesson/types';
import { RetakeConfirmationModal } from '@dc/components/Student/Lesson/Survey/RetakeConfirmationModal/RetakeConfirmationModal';

import { ReactComponent as RefreshIcon } from '@shared/svg/refresh.svg';
import { ReactComponent as EyeIcon } from '@shared/svg/eye.svg';
import SharedButton from '@shared/components/Button/Button';
import { callToast } from '@shared/components/Toaster/Toaster';
import { IconContainer } from '@shared/components/IconContainer/IconContainer';
import { Tooltip } from '@shared/components/Tooltip';
import { useToggle } from '@shared/hooks/useToggle';
import { PartiallyOptional } from '@shared/utils/types';

import Card from '../shared/Card/Card';

type Props = {
  careerReviewSurvey: PartiallyOptional<TCareerReviewSurvey, 'performed' | 'version'> | null;
  courseName: string;
  previousItemsCompleted?: boolean;
  previewOnly?: boolean;
  resourceId: string;
};

export type SurveyFormType = Record<
  string,
  {
    answers: string[] | string;
    otherAnswer: string;
  }
>;

export const CareerReviewSurvey = ({
  careerReviewSurvey,
  courseName,
  resourceId,
  previewOnly,
  previousItemsCompleted,
}: Props) => {
  const { courseId, opportunityId } = useParams<{ courseId: string; opportunityId: string }>();
  const history = useHistory();
  const { t } = useTranslation();
  const [isRetakeConfirmationModalOpen, toggleIsRetakeConfirmationModalOpen] = useToggle(false);

  if (!careerReviewSurvey) return null;

  const { questions } = careerReviewSurvey;

  const virtualInternshipRefetchQueries = [
    { query: OPPORTUNITY_QUERY, variables: { id: opportunityId } },
    { query: VIRTUAL_INTERNSHIP_CONTENT_QUERY, variables: { opportunityId } },
  ];

  const courseRefetchQuery = [{ query: currentCoursesQuery }];

  const refetchQueries = courseId ? courseRefetchQuery : virtualInternshipRefetchQueries;

  const [createCareerReviewSurveyAnswers, { loading }] = useCreateCareerReviewSurveyAnswers({
    refetchQueries,
  });

  const initialValues: SurveyFormType = questions.reduce((acc, { id, type }) => {
    const answers = type === 'single_choice' ? '' : [];

    return { ...acc, [id]: { answers, otherAnswer: '' } };
  }, {});

  const onSuccess = () => {
    if (previousItemsCompleted) {
      const location = courseId
        ? {
            pathname: `/courses/${courseId}/complete`,
            state: { courseName },
          }
        : `/opportunities/${opportunityId}/virtual-internship`;

      history.push(location);
    } else {
      history.goBack();
    }
  };

  const handleSubmit = (values: SurveyFormType) => {
    const hasAllAnswers = Object.values(values).every((question) => {
      const hasAnyAnswers = question.answers.length > 0;

      if (question.answers.includes('Other')) {
        return hasAnyAnswers && !isEmpty(question.otherAnswer);
      }

      return hasAnyAnswers;
    });

    if (!hasAllAnswers) {
      callToast('error', t('careerReviewSurvey.notAllAnswersNotification'));

      return;
    }

    const payload = map(values, (question, questionId) => {
      let answers = castArray(question.answers).filter((answer) => answer !== 'Other');

      if (question.answers.includes('Other') && !isEmpty(question?.otherAnswer)) {
        answers.push(question?.otherAnswer);
      }

      return {
        answer: answers,
        questionId: String(questionId),
      };
    });

    const context = courseId ? 'COURSE' : 'VIRTUAL_INTERNSHIP';

    createCareerReviewSurveyAnswers({
      contextId: resourceId,
      contextType: CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES[context],
      answers: payload,
    })
      .then(() => {
        callToast('success', t('careerReviewSurvey.completedSuccessfullyNotification'));
        onSuccess();
      })
      .catch((e) => {
        if (e instanceof ApolloError) callToast('error', e.message);
      });
  };

  const summaryMode = careerReviewSurvey.performed;

  const retakeAvailable = summaryMode && !previewOnly;

  const contextType = history.location.pathname.includes('virtual-internship')
    ? CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES.VIRTUAL_INTERNSHIP
    : CAREER_REVIEW_SURVEY_ANSWER_CONTEXT_TYPES.COURSE;

  return (
    <div className='flex flex-col gap-base xxxl:gap-md'>
      <Card className='!mb-0'>
        <div className='flex justify-between items-center mb-md'>
          <div className='flex w-full'>
            <IconContainer
              Icon={TableOfContentIcon}
              className='bg-primary-200 rounded-sm mr-sm text-primary-500'
            />
            <div className='text-800 border-l-2 border-neutral-200 pl-sm'>
              <span className='leading-lg text-xs'>{courseName}</span>
              <h5 className='leading-lg text-sm mb-0'>{t('careerReviewSurvey.surveyTitle')}</h5>
            </div>
            {retakeAvailable && (
              <SharedButton
                Icon={RefreshIcon}
                className='ms-auto'
                iconPlacement='start'
                variant='primary-outlined'
                onClick={toggleIsRetakeConfirmationModalOpen}>
                {t('careerReviewSurvey.retake')}
              </SharedButton>
            )}
          </div>
          {previewOnly && (
            <div className='py-xxs px-xs border border-secondary-500 rounded-full min-w-fit'>
              <Tooltip
                message={t('careerReviewSurvey.surveyPreviewModeInfo')}
                side='bottom'
                sideOffset={8}>
                <div className='flex items-center text-secondary-500 font-medium'>
                  <IconContainer Icon={EyeIcon} className='mr-xxs' paddingSize='none' size='sm' />
                  <p className='mb-0 leading-lg'>{t('careerReviewSurvey.surveyPreviewMode')}</p>
                </div>
              </Tooltip>
            </div>
          )}
        </div>
        {summaryMode && <SurveySummary questions={questions} />}
        {!summaryMode && (
          <Formik initialValues={initialValues} onSubmit={handleSubmit}>
            <SurveyForm
              isSaving={loading}
              previewOnly={previewOnly}
              questions={questions}
              version={careerReviewSurvey.version}
            />
          </Formik>
        )}
      </Card>
      {summaryMode && (
        <SharedButton className='ml-auto' variant='primary' onClick={onSuccess}>
          {previousItemsCompleted
            ? t('careerReviewSurvey.endCourse')
            : t('careerReviewSurvey.exitSurvey')}
        </SharedButton>
      )}
      {retakeAvailable && (
        <RetakeConfirmationModal
          contextId={resourceId}
          contextType={contextType}
          isOpen={isRetakeConfirmationModalOpen}
          onDismiss={toggleIsRetakeConfirmationModalOpen}
        />
      )}
    </div>
  );
};
