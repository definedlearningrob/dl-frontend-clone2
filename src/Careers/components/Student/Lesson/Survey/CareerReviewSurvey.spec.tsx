import { waitFor, screen, within } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import userEvent from '@testing-library/user-event';
import { Route } from 'react-router-dom';

import { renderWithRouterAndReduxProvider } from '@dc/utils/test';
import { CareerReviewSurvey } from '@dc/components/Student/Lesson/Survey/CareerReviewSurvey';

import {
  createAttemptSpy,
  createCareerReviewSurveyAnswersSpy,
  currentCoursesSpy,
  lessonInCourseSpy,
  mocks,
} from './mocks';

type GetCareerReviewSurveyMockInput = {
  performed?: boolean;
  answers?: string[][];
};

const getCareerReviewSurveyMock = ({
  performed = false,
  answers = [],
}: GetCareerReviewSurveyMockInput) => ({
  version: 2 as const,
  performed,
  questions: [
    {
      answer: answers[0] || [],
      id: '1',
      options: [
        {
          option: 'Yes, definitely',
          step: 1,
        },
        {
          option: 'Definitely not',
          step: 2,
        },
      ],
      question: 'Can blind people see their dreams?',
      type: 'single_choice',
    },
    {
      answer: answers[1] || [],
      id: '2',
      options: [
        {
          option: 'Very Interested',
          step: 1,
        },
        {
          option: 'Not at all Interested',
          step: 2,
        },
        { option: 'Other', step: 3 },
      ],
      question: 'How interested are you in the Pathway this course is a part of?',
      type: 'multiple_choice',
    },
  ],
});

const renderSurvey = (
  mocks: MockedResponse[] = [],
  previewOnly = false,
  careerReviewSurvey: any
) => {
  const lessonMock = {
    careerReviewSurvey,
    resourceId: '1',
  };

  return renderWithRouterAndReduxProvider(
    <MockedProvider mocks={mocks}>
      <Route path='courses/:courseId/lessons/:lessonId'>
        <CareerReviewSurvey
          careerReviewSurvey={careerReviewSurvey}
          courseName='Test course name'
          previewOnly={previewOnly}
          resourceId={lessonMock.resourceId}
        />
      </Route>
    </MockedProvider>,
    {
      route: `courses/2/lessons/111`,
    }
  );
};

describe('Career Review Survey', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders correctly as a form', () => {
    const { container } = renderSurvey(mocks, false, getCareerReviewSurveyMock({}));

    expect(container).toMatchSnapshot();
  });

  it('renders correctly as a summary', () => {
    const { container } = renderSurvey(mocks, true, getCareerReviewSurveyMock({}));

    expect(container).toMatchSnapshot();
  });

  it('renders correctly as a summary if survey already performed', () => {
    const answers = [['Yes, definitely'], ['Very Interested', 'Not at all Interested']];

    const { container } = renderSurvey(
      mocks,
      true,
      getCareerReviewSurveyMock({ performed: true, answers })
    );

    expect(container).toMatchSnapshot();
  });

  it('should allow user to retake survey', async () => {
    const answers = [['Yes, definitely'], ['Very Interested', 'Not at all Interested']];

    renderSurvey(mocks, false, getCareerReviewSurveyMock({ performed: true, answers }));

    const retakeButton = await screen.findByRole('button', { name: 'Retake' });

    userEvent.click(retakeButton);

    const confirmationModal = await screen.findByRole('dialog', { name: 'Modal' });

    const modalHeading = screen.getByRole('heading', {
      name: 'Do you want to retake Career Review Survey?',
      level: 4,
    });

    expect(modalHeading).toBeInTheDocument();

    const confirmButton = within(confirmationModal).getByRole('button', { name: 'Retake' });

    userEvent.click(confirmButton);

    await waitFor(() => expect(createAttemptSpy).toHaveBeenCalledTimes(1));

    await waitFor(() => expect(lessonInCourseSpy).toHaveBeenCalledTimes(1));
  });

  it('disables save survey until all questions are not answered', () => {
    renderSurvey(mocks, false, getCareerReviewSurveyMock({}));

    const saveButton = screen.getByRole('button', { name: 'Save' });
    expect(saveButton).not.toBeDisabled();
  });

  it('enables save survey correctly when all questions are answered', async () => {
    renderSurvey(
      mocks,
      false,
      getCareerReviewSurveyMock({
        performed: false,
      })
    );
    const firstAnswerOfSingleChoiceQuestion = screen.getByRole('radio', {
      name: 'Yes, definitely',
    });

    const firstAnswerOfMultiChoiceQuestion = screen.getByRole('checkbox', {
      name: 'Very Interested',
    });

    const saveButton = screen.getByRole('button', { name: 'Save' });

    userEvent.click(firstAnswerOfSingleChoiceQuestion);
    userEvent.click(firstAnswerOfMultiChoiceQuestion);

    userEvent.click(saveButton);

    await waitFor(() => {
      expect(createCareerReviewSurveyAnswersSpy).toBeCalledTimes(1);
    });
    await waitFor(() => {
      expect(currentCoursesSpy).toBeCalledTimes(1);
    });
  });

  it('enables select only one answer in single choice question', async () => {
    renderSurvey(
      mocks,
      false,
      getCareerReviewSurveyMock({
        performed: false,
      })
    );
    const [singleChoiceOption1, singleChoiceOption2] = await screen.findAllByRole('radio');

    userEvent.click(singleChoiceOption1);
    await waitFor(() => {
      expect(singleChoiceOption1).toBeChecked();
    });

    userEvent.click(singleChoiceOption2);
    await waitFor(() => {
      expect(singleChoiceOption2).toBeChecked();
    });

    expect(singleChoiceOption1).not.toBeChecked();

    const saveButton = screen.getByRole('button', { name: 'Save' });
    userEvent.click(saveButton);

    await waitFor(() => {
      expect(createCareerReviewSurveyAnswersSpy).toBeCalledTimes(0);
    });
    await waitFor(() => {
      expect(currentCoursesSpy).toBeCalledTimes(0);
    });
  });

  it('enables select many answers in multi choice question', async () => {
    renderSurvey(
      mocks,
      false,
      getCareerReviewSurveyMock({
        performed: false,
      })
    );
    const [multiChoiceOption1, multiChoiceOption2] = await screen.findAllByRole('checkbox');

    userEvent.click(multiChoiceOption1);
    await waitFor(() => {
      expect(multiChoiceOption1).toBeChecked();
    });

    userEvent.click(multiChoiceOption2);
    await waitFor(() => {
      expect(multiChoiceOption1).toBeChecked();
    });

    await waitFor(() => {
      expect(createCareerReviewSurveyAnswersSpy).toBeCalledTimes(0);
    });
    expect(currentCoursesSpy).toBeCalledTimes(0);
  });

  it('when first answer is selected in multi choice question, then other answers are disabled', async () => {
    renderSurvey(
      mocks,
      false,
      getCareerReviewSurveyMock({
        performed: false,
      })
    );

    const multiChoiceOption1 = screen.getByRole('checkbox', { name: 'Very Interested' });
    const multiChoiceOption2 = screen.getByRole('checkbox', { name: 'Not at all Interested' });
    const multiChoiceOption3 = screen.getByRole('checkbox', { name: 'Other' });

    expect(multiChoiceOption2).not.toBeDisabled();

    userEvent.click(multiChoiceOption1);
    await waitFor(() => {
      expect(multiChoiceOption1).toBeChecked();
    });

    expect(multiChoiceOption2).toBeDisabled();
    expect(multiChoiceOption3).toBeDisabled();

    await waitFor(() => {
      expect(createCareerReviewSurveyAnswersSpy).toBeCalledTimes(0);
    });
    expect(currentCoursesSpy).toBeCalledTimes(0);
  });

  it('when user selects "Other" option in multi choice question, then text area is enabled', async () => {
    renderSurvey(
      mocks,
      false,
      getCareerReviewSurveyMock({
        performed: false,
      })
    );

    const multiChoiceOption3 = screen.getByRole('checkbox', { name: 'Other' });

    userEvent.click(multiChoiceOption3);
    await waitFor(() => {
      expect(multiChoiceOption3).toBeChecked();
    });

    const textArea = screen.getByRole('textbox');

    expect(textArea).toBeEnabled();

    await waitFor(() => {
      expect(createCareerReviewSurveyAnswersSpy).toBeCalledTimes(0);
    });
    expect(currentCoursesSpy).toBeCalledTimes(0);
  });
});
