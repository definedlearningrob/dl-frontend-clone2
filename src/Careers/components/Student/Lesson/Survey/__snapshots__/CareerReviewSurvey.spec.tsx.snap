// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Career Review Survey renders correctly as a form 1`] = `
<div>
  <div
    class="flex flex-col gap-base xxxl:gap-md"
  >
    <div
      class="lesson-item-card !mb-0"
    >
      <div
        class="flex justify-between items-center mb-md"
      >
        <div
          class="flex w-full"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0 bg-primary-200 rounded-sm mr-sm text-primary-500 p-xs"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
          <div
            class="text-800 border-l-2 border-neutral-200 pl-sm"
          >
            <span
              class="leading-lg text-xs"
            >
              Test course name
            </span>
            <h5
              class="leading-lg text-sm mb-0"
            >
              Career Review Survey
            </h5>
          </div>
        </div>
      </div>
      <form
        action="#"
        class="form"
      >
        <div
          class="mb-base"
        >
          <p
            class="font-bold leading-base text-xs xxxl:text-base mb-sm"
          >
            1. Can blind people see their dreams?
          </p>
          <ul
            role="group"
          >
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <label
                class="radioLabel border border-neutral-300 hover:border-neutral-600 text-neutral-700 rounded-xs cursor-pointer flex items-center leading-lg font-medium p-sm xxxl:p-base"
              >
                <input
                  class="radioInput"
                  name="Yes, definitely"
                  type="radio"
                  value="Yes, definitely"
                />
                <div
                  class="radioSpan"
                >
                  <span
                    class="labelWrapper"
                  >
                    Yes, definitely
                  </span>
                </div>
              </label>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <label
                class="radioLabel border border-neutral-300 hover:border-neutral-600 text-neutral-700 rounded-xs cursor-pointer flex items-center leading-lg font-medium p-sm xxxl:p-base"
              >
                <input
                  class="radioInput"
                  name="Definitely not"
                  type="radio"
                  value="Definitely not"
                />
                <div
                  class="radioSpan"
                >
                  <span
                    class="labelWrapper"
                  >
                    Definitely not
                  </span>
                </div>
              </label>
            </li>
          </ul>
        </div>
        <div
          class="mb-base"
        >
          <p
            class="font-bold leading-base text-xs xxxl:text-base mb-sm"
          >
            2. How interested are you in the Pathway this course is a part of?
          </p>
          <ul
            role="group"
          >
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  id="checkbox-2.answers-Very Interested"
                  tabindex="0"
                  type="checkbox"
                  value="Very Interested"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Very Interested"
                >
                  Very Interested
                </label>
              </div>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  id="checkbox-2.answers-Not at all Interested"
                  tabindex="0"
                  type="checkbox"
                  value="Not at all Interested"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Not at all Interested"
                >
                  Not at all Interested
                </label>
              </div>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  id="checkbox-2.answers-Other"
                  tabindex="0"
                  type="checkbox"
                  value="Other"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Other"
                >
                  Other
                </label>
              </div>
            </li>
          </ul>
        </div>
        <button
          class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] ml-auto -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
          data-testid="button"
          type="submit"
        >
          <span
            class="items-center flex justify-center gap-xs"
          >
            Save
          </span>
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`Career Review Survey renders correctly as a summary 1`] = `
<div>
  <div
    class="flex flex-col gap-base xxxl:gap-md"
  >
    <div
      class="lesson-item-card !mb-0"
    >
      <div
        class="flex justify-between items-center mb-md"
      >
        <div
          class="flex w-full"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0 bg-primary-200 rounded-sm mr-sm text-primary-500 p-xs"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
          <div
            class="text-800 border-l-2 border-neutral-200 pl-sm"
          >
            <span
              class="leading-lg text-xs"
            >
              Test course name
            </span>
            <h5
              class="leading-lg text-sm mb-0"
            >
              Career Review Survey
            </h5>
          </div>
        </div>
        <div
          class="py-xxs px-xs border border-secondary-500 rounded-full min-w-fit"
        >
          <span
            data-state="closed"
          >
            <div
              class="flex items-center text-secondary-500 font-medium"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 mr-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <p
                class="mb-0 leading-lg"
              >
                Preview Mode
              </p>
            </div>
          </span>
        </div>
      </div>
      <form
        action="#"
        class="form"
      >
        <div
          class="mb-base"
        >
          <p
            class="font-bold leading-base text-xs xxxl:text-base mb-sm"
          >
            1. Can blind people see their dreams?
          </p>
          <ul
            role="group"
          >
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <label
                class="radioLabel border border-neutral-300 hover:border-neutral-600 text-neutral-700 rounded-xs cursor-pointer flex items-center leading-lg font-medium p-sm xxxl:p-base bg-neutral-200 !cursor-not-allowed !cursor-not-allowed"
              >
                <input
                  class="radioInput"
                  disabled=""
                  name="Yes, definitely"
                  type="radio"
                  value="Yes, definitely"
                />
                <div
                  class="radioSpan"
                >
                  <span
                    class="labelWrapper"
                  >
                    Yes, definitely
                  </span>
                </div>
              </label>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <label
                class="radioLabel border border-neutral-300 hover:border-neutral-600 text-neutral-700 rounded-xs cursor-pointer flex items-center leading-lg font-medium p-sm xxxl:p-base bg-neutral-200 !cursor-not-allowed !cursor-not-allowed"
              >
                <input
                  class="radioInput"
                  disabled=""
                  name="Definitely not"
                  type="radio"
                  value="Definitely not"
                />
                <div
                  class="radioSpan"
                >
                  <span
                    class="labelWrapper"
                  >
                    Definitely not
                  </span>
                </div>
              </label>
            </li>
          </ul>
        </div>
        <div
          class="mb-base"
        >
          <p
            class="font-bold leading-base text-xs xxxl:text-base mb-sm"
          >
            2. How interested are you in the Pathway this course is a part of?
          </p>
          <ul
            role="group"
          >
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer bg-neutral-200"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  disabled=""
                  id="checkbox-2.answers-Very Interested"
                  tabindex="0"
                  type="checkbox"
                  value="Very Interested"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Very Interested"
                >
                  Very Interested
                </label>
              </div>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer bg-neutral-200"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  disabled=""
                  id="checkbox-2.answers-Not at all Interested"
                  tabindex="0"
                  type="checkbox"
                  value="Not at all Interested"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Not at all Interested"
                >
                  Not at all Interested
                </label>
              </div>
            </li>
            <li
              class="list-none mb-sm last-of-type:mb-0"
            >
              <div
                class="checkbox border border-neutral-300 hover:border-neutral-600 !text-neutral-700 font-medium leading-lg rounded-xs cursor-pointer bg-neutral-200"
              >
                <input
                  class="checkbox__input"
                  data-testid="checkbox"
                  disabled=""
                  id="checkbox-2.answers-Other"
                  tabindex="0"
                  type="checkbox"
                  value="Other"
                />
                <label
                  class="checkbox__label !p-sm xxxl:!p-base after:!left-sm xxxl:after:!left-base"
                  for="checkbox-2.answers-Other"
                >
                  Other
                </label>
              </div>
            </li>
          </ul>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`Career Review Survey renders correctly as a summary if survey already performed 1`] = `
<div>
  <div
    class="flex flex-col gap-base xxxl:gap-md"
  >
    <div
      class="lesson-item-card !mb-0"
    >
      <div
        class="flex justify-between items-center mb-md"
      >
        <div
          class="flex w-full"
        >
          <div
            class="flex"
          >
            <div
              class="flex align-center justify-center grow-0 shrink-0 bg-primary-200 rounded-sm mr-sm text-primary-500 p-xs"
              data-testid="icon"
            >
              <div
                class="h-base w-base"
              />
            </div>
          </div>
          <div
            class="text-800 border-l-2 border-neutral-200 pl-sm"
          >
            <span
              class="leading-lg text-xs"
            >
              Test course name
            </span>
            <h5
              class="leading-lg text-sm mb-0"
            >
              Career Review Survey
            </h5>
          </div>
        </div>
        <div
          class="py-xxs px-xs border border-secondary-500 rounded-full min-w-fit"
        >
          <span
            data-state="closed"
          >
            <div
              class="flex items-center text-secondary-500 font-medium"
            >
              <div
                class="flex"
              >
                <div
                  class="flex align-center justify-center grow-0 shrink-0 mr-xxs"
                  data-testid="icon"
                >
                  <div
                    class="h-sm w-sm"
                  />
                </div>
              </div>
              <p
                class="mb-0 leading-lg"
              >
                Preview Mode
              </p>
            </div>
          </span>
        </div>
      </div>
      <div
        class="mb-base"
      >
        <h4
          class="mb-xs"
        >
          Previous Answers
        </h4>
        <p
          class="text-xs xxxl:text-sm leading-lg"
        >
          Here are your answers, in order to change them please retake the whole survey
        </p>
      </div>
      <div
        class="rounded-sm p-sm border border-neutral-300 bg-neutral-200 mb-base last-of-type:mb-0"
      >
        <h6
          class="mb-0 mb-xs"
        >
          1. Can blind people see their dreams?
        </h6>
        <ul
          class="list-disc pl-base marker:text-neutral-700 text-xs"
        >
          <li
            class="leading-lg"
          >
            Yes, definitely
          </li>
        </ul>
      </div>
      <div
        class="rounded-sm p-sm border border-neutral-300 bg-neutral-200 mb-base last-of-type:mb-0"
      >
        <h6
          class="mb-0 mb-xs"
        >
          2. How interested are you in the Pathway this course is a part of?
        </h6>
        <ul
          class="list-disc pl-base marker:text-neutral-700 text-xs"
        >
          <li
            class="leading-lg"
          >
            Very Interested
          </li>
          <li
            class="leading-lg"
          >
            Not at all Interested
          </li>
        </ul>
      </div>
    </div>
    <button
      class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] ml-auto -primary bg-primary-500 hover:bg-primary-600 outline-primary-500 border-primary-500 py-xs px-base leading-[22px] border text-white"
      data-testid="button"
      type="button"
    >
      <span
        class="items-center flex justify-center gap-xs"
      >
        Exit Career Survey
      </span>
    </button>
  </div>
</div>
`;
